### DISCUSSION TITLE
Animated / Rigged geometry not showing up in QuickLook.

### DISCUSSION THREAD

**Discussion by samsloan** [google/model-viewer#4538] [Q&A] [answered]
Hi Everyone,

I have come across the issue where the USDZ convert seems to be culling any geometry in my scene that has been rigged/animated. The animation actions work happily in model-viewer itself and everything behaves nicely, however, in QuickLook any geometry in my armature hierarchy from Blender is culled.

I'm assuming this is an issue with the USDZ convert and I can just do my own USDZ convert and use ios-src, but I would not be able to keep the current variant state (eg. seat colors) from model-viewer into QuickLook.

I just wanted to check if anyone was aware of any workarounds before I abandon hopes of being able to bridge the variant state into QuickLook.

The two images attached are screenshots of the model in QuickLook. One rigged and one not (obviously the one with less geometry is the one that is rigged)

Cheers!
Sam
![395692920_1031586514848890_1922979295240179651_n](https://github.com/google/model-viewer/assets/18361357/2759c712-38e9-476e-b423-b85111dcc57e)
![395174039_1097462848284531_266003063838276711_n](https://github.com/google/model-viewer/assets/18361357/fb24e076-45e6-485b-aca1-8feab523d474)

**Comment by elalish** [comment, marked as answer]
Yes, this is a known limitation of the three.js USDZExporter, which is what does our conversion. If you're interested in contributing upstream, I'm sure many users (and us!) would be very appreciative.

**Comment by elalish** [comment]
FYI @mrdoob
